amethyst_renderer/src/pipe/effect/mod.rs

```diff
@@ -94,8 +90,11 @@ impl Effect {
         match self.const_bufs.get(name.as_ref()) {
             Some(i) => {
                 let raw = &self.data.const_bufs[*i];
-                enc.update_buffer::<T>(unsafe { mem::transmute(raw) }, &data[..], 0)
-                    .expect("Failed to update buffer (TODO: replace expect)");
+                enc.update_buffer::<T>(
+                    unsafe { &*(raw as *const RawBuffer<_> as *const Buffer<_, _>) },
```

Comment:
I can't read this, so I'd like to know if this was reviewed by the other reviewers.

Reply:
Yes, it was ;)